### ISSUE TITLE
Inserting JSON data to R function

### ISSUE THREAD

**Issue by kamranesmaeili** [FabrizioSandri/r-integration#1]
Is there any way to input keys and values to R instead of only values? Consider the following example:

```
#In test.R
my_val <- function(a, b, c, d){
     return(a*b*c*d)
}
```

The following works fine but the input values must be inserted in order of the function "my_val":
```
#In nodejs
const R = require('r-integration');
let result = R.callMethod("./test.R", "my_val", ["2", "3", "4", "5"]);
console.log(result);
```
But I cannot insert the following to be read inside R, in case the inputs are not inserted in the right order:
```
#In nodejs
const R = require('r-integration');
let result = R.callMethod("./test.R", "my_val", ["d": "5", "b":"3", "c":"4", "a":"2"]);
console.log(result);
```
Any suggestions?

**Comment by FabrizioSandri** [comment]
Good catch !

I just released a new version 1.2.0 which supports complex parameters. 
Now you can pass parameters to functions by using an `Object` or an `Array`.

In your example
```js
// In NodeJS
const R = require('r-integration');

let result = R.callMethod("./test.R", "my_val", {d: 5, b: 3, c: 4, a: 2});

console.log(result);

> [ '120' ]
```

Documentation: [ Call a R function with parameters](https://www.npmjs.com/package/r-integration)

**Comment by kamranesmaeili** [comment]
Version 1.2.0 works really well, thank you!

**Comment by steven-so** [comment]
@FabrizioSandri @kamranesmaeili 

What if you are trying to pass a Javascript array into the R function? I'm getting an unused argument error: 
```
[R: compile error] Error in my_val(a = 43.28, 21.64, 32.46, 64.92, 43.28, 10.82, 10.82, b = 10.82,  :
  unused arguments (21.64, 32.46, 64.92, 43.28, 10.82, 10.82, 21.64, 54.1, 32.46, 32.46, 10.82, 32.46)
Calls: print
Execution halted
```

javascript: 
```
    let result = R.callMethod("./rScripts/welchs.R", "my_val",
    {
      a: ['43.28', '21.64', '32.46', '64.92', '43.28', '10.82', '10.82'],
      b: ['10.82', '21.64', '54.10', '32.46', '32.46', '10.82', '32.46'],
    });
```


the welchsR script: 
```
my_val <- function(a, b){

    data1 <- as.numeric(c("123,123,421"))
    data2 <- as.numeric(c("32,123,123"))

    result <- t.test(data1, data2, alternative="two.sided", var.equal=FALSE)
    
    return(result)
}
```

**Comment by FabrizioSandri** [comment]
> @FabrizioSandri @kamranesmaeili
> 
> What if you are trying to pass a Javascript array into the R function? I'm getting an unused argument error:
> 
> ```
> [R: compile error] Error in my_val(a = 43.28, 21.64, 32.46, 64.92, 43.28, 10.82, 10.82, b = 10.82,  :
>   unused arguments (21.64, 32.46, 64.92, 43.28, 10.82, 10.82, 21.64, 54.1, 32.46, 32.46, 10.82, 32.46)
> Calls: print
> Execution halted
> ```
> 
> javascript:
> 
> ```
>     let result = R.callMethod("./rScripts/welchs.R", "my_val",
>     {
>       a: ['43.28', '21.64', '32.46', '64.92', '43.28', '10.82', '10.82'],
>       b: ['10.82', '21.64', '54.10', '32.46', '32.46', '10.82', '32.46'],
>     });
> ```
> 
> the welchs.R script:
> 
> ```
> my_val <- function(a, b){
> 
>     data1 <- as.numeric(c(a))
>     data2 <- as.numeric(c(b))
> 
>     result <- t.test(data1, data2, alternative="two.sided", var.equal=FALSE)
>     
>     return(result)
> }
> ```

Good catch! 
Just released a new version 1.3.0 which supports arrays as parameter.

Update to the new version by typing
`npm install r-integration`

**Comment by steven-so** [comment]
@FabrizioSandri 

Thanks that's awesome!
